components/job-orchestration/job_orchestration/scheduler/compress/compression_scheduler.py

```diff
@@ -162,6 +193,38 @@ def search_and_schedule_new_tasks(db_conn, db_cursor, clp_metadata_db_connection
         clp_io_config = ClpIoConfig.parse_obj(
             msgpack.unpackb(brotli.decompress(job_row["clp_config"]))
         )
+        input_config = clp_io_config.input
+
+        table_prefix = clp_metadata_db_connection_config["table_prefix"]
+        if StorageEngine.CLP_S == clp_storage_engine:
+            dataset_name = input_config.dataset
+            if dataset_name not in datasets_cache:
+                archive_storage_directory: Path
+                if StorageType.S3 == clp_archive_output.storage.type:
+                    s3_config = clp_archive_output.storage.s3_config
+                    archive_storage_directory = Path(s3_config.key_prefix) / dataset_name
+                else:
+                    archive_storage_directory = clp_archive_output.get_directory() / dataset_name
+
+                try:
+                    insert_new_datasets_table_entry(
+                        db_cursor,
+                        table_prefix,
+                        dataset_name,
+                        clp_archive_output.storage.type,
+                        archive_storage_directory,
+                    )
+                    create_metadata_db_tables(db_cursor, table_prefix, dataset_name)
+                    db_conn.commit()
+                except Exception as e:
```

Comment:
What exception are you anticipating here?

Reply:
Just generally if any SQL operation happens to fail, or if in the future there will be two schedulers racing to register datasets.